contrib/pkg/awstagdeprovision/awstagdeprovision.go

```diff
@@ -108,7 +124,9 @@ func (o *ClusterUninstaller) Run() error {
 
 	// launch goroutines
 	for name, function := range deleteFuncs {
-		go deleteRunner(name, function, awsSession, o.Filters, o.ClusterName, o.Logger, returnChannel)
+		for _, filter := range o.Filters {
+			go deleteRunner(name, function, awsSession, filter, o.ClusterName, o.Logger, returnChannel)
+		}
 	}
 
 	// wait for them to finish
```

Comment:
need to increase the number of goroutines we're waiting to complete now that we launch per-key=value.

```suggestion
	// wait for them to finish
       for i := 0; i < len(deleteFuncs) * len(o.Filters); i++ {
```

Reply:
> need to increase the number of goroutines we're waiting to complete...

Good catch :).  Fixed with 7522689 -> b1cad98.